hax/hax/motr/__init__.py

```diff
@@ -355,19 +357,31 @@ def ha_obj_state(st):
                                                             kv_cache=kv_cache)
         if not notes:
             return []
+        notes_pending = notes
+        notes_len_total = len(notes)
+        idx = 0
         message_ids: List[MessageId] = []
-        LOG.debug('broadcast ha states phase %s', broadcast_hax_only)
-        if broadcast_hax_only:
-            hax_endpoint = self.consul_util.get_hax_endpoint()
-            message_ids = self._ffi.ha_broadcast_hax_only(
-                self._ha_ctx, make_array(HaNoteStruct, notes), len(notes),
-                make_c_str(hax_endpoint))
-        else:
-            message_ids = self._ffi.ha_broadcast(
-                self._ha_ctx, make_array(HaNoteStruct, notes), len(notes))
-        LOG.debug(
-            'Broadcast HA state complete with the following message_ids = %s',
-            message_ids)
+        while notes_pending:
+            if len(notes_pending) > MAX_MOTR_NVEC_UPDATE_SZ:
+                notes_to_send = notes[idx:(idx + MAX_MOTR_NVEC_UPDATE_SZ)]
+            else:
+                notes_to_send = notes[idx:notes_len_total]
+            notes_to_send_len = len(notes_to_send)
+            if broadcast_hax_only:
+                hax_endpoint = self.consul_util.get_hax_endpoint()
+                message_ids = self._ffi.ha_broadcast_hax_only(
+                    self._ha_ctx, make_array(HaNoteStruct, notes_to_send),
+                    notes_to_send_len,
+                    make_c_str(hax_endpoint))
+            else:
+                message_ids = self._ffi.ha_broadcast(
+                    self._ha_ctx, make_array(HaNoteStruct, notes_to_send),
+                    notes_to_send_len)
+            LOG.debug(
+                'Broadcast HA state complete, message_ids = %s',
+                message_ids)
+            idx += notes_to_send_len
+            notes_pending = notes[idx:notes_len_total]
```

Comment:
[Optional] The method is 108 lines long. Perhaps it is time to refactor it and split into smaller functions.

Reply:
Yes, makes sense, let me do that.